src/util/list.h

```diff
@@ -113,6 +125,25 @@ list_node_t *search_list(list_t *list, int key);
 */
 void remove_node(list_t *list, int key);
 
+//---------------------------//
+//  Data element retrievers  //
+//---------------------------//
+
+/**
+@brief Returns pointer to data element of first list item.
+*/
+void *get_first_data(list_t *list);
```

Comment:
The interface provided in the previous implementation worked just fine. I don't see the value in going to all this trouble to hide `list_node_t`.

Reply:
Because my approach is cleaner, more robust and simpler to understand. That's what refactoring is all about.